tests/test_planning.py

```diff
@@ -54,24 +57,34 @@ def test_task_plan():
     heuristic = utils.create_task_planning_heuristic("hadd", init_atoms,
                                                      task.goal, ground_nsrts,
                                                      env.predicates, objects)
-    skeleton, _, _ = task_plan(init_atoms,
-                               task.goal,
-                               ground_nsrts,
-                               reachable_atoms,
-                               heuristic,
-                               timeout=1,
-                               seed=123)
+    task_plan_generator = task_plan(init_atoms,
+                                    task.goal,
+                                    ground_nsrts,
+                                    reachable_atoms,
+                                    heuristic,
+                                    timeout=1,
+                                    seed=123)
+    skeleton, _, metrics = next(task_plan_generator)
     assert len(skeleton) == 2
     assert isinstance(skeleton[0], _GroundNSRT)
     assert isinstance(skeleton[1], _GroundNSRT)
+    total_num_skeletons = 1
+    with pytest.raises(ApproachFailure) as e:
+        for _, _, metrics in task_plan_generator:
+            total_num_skeletons += 1
+            assert metrics["num_skeletons_optimized"] == total_num_skeletons
+    assert "Planning reached max_skeletons_optimized!" in str(e.value)
```

Comment:
i'm confused, why is this assert getting hit? is it because we don't pass a max_num into task_plan? should we be?